### DISCUSSION TITLE
AWS EBS not expanding

### DISCUSSION THREAD

**Discussion by balthazar** [topolvm/pvc-autoresizer#171] [Q&A] [answered]
I've tried without success to understand why the autoresizer would not resize my filled PVCs.

I'm using EKS, EBS volumes (EBS CSI plugin) with a Storage Class allowing for volume expansion. The PVCs are tagged with the proper annotations, Prometheus is running and properly reporting the kubelet volume stats.

![1674865703](https://user-images.githubusercontent.com/6033345/215229862-9268229f-2fbd-49f2-baf5-1022ff00fa38.png)

The `pvc-autoresizer` pod give me the following metrics:

```
pvcautoresizer_kubernetes_client_fail_total 0
pvcautoresizer_loop_seconds_total 0.7255873060000002
pvcautoresizer_metrics_client_fail_total 0
```

And properly elected itself as the leader, but without giving any further logs

```
leaderelection.go:248] attempting to acquire leader lease pvc-autoresizer/49e22f61.topolvm.io...                                           
leaderelection.go:258] successfully acquired lease pvc-autoresizer/49e22f61.topolvm.io
```

So I'm really not sure what I'm doing wrong! Apologies if it seems obvious.

PS: Is it normal that [this file](https://github.com/topolvm/pvc-autoresizer/blob/main/metrics/mertics.go) is named mertics? I'm assuming it's a typo!
PSS: I can't seem to be able to join the Slack channel, otherwise I would have posted there instead

**Comment by daichimukai** [comment, marked as answer]
@balthazar 
I have a few questions and requests.

- What version of pvc-autoresizer are you using?
- Did you install pvc-autoresizer using helm?
- Please share the manifest of the PVC and its storage class that is not resizing as expected.

**Reply by balthazar** [reply]
Oh god I'm so stupid, I forgot the `storage_limit` annotation... Thanks for your help!
